### ISSUE TITLE
[feat] MacOSX Support for tauri-driver

### ISSUE THREAD

**Issue by prestomation** [tauri-apps/tauri#7068]
### Describe the problem

tauri-driver does not support Mac.

I have a Tauri app that has some platform-specific code in it. (Think: tauri-plugin-deep-link).
I need to write automated tests of my app, and I need to run on every platform, as the platform-specific code must be verified on every platform.

### Describe the solution you'd like

I would like tauri-driver to support mac similarly to how it supports windows and linux

### Alternatives considered

Selenium Test on windows/linux, and get by with manual testing and unit tests on mac for mac-specific functionality

### Additional context

I understand this is not a priority for the tauri team, understandable as most app testing can be done on a single platform.
I still wanted to file this to collect interest and any more information about solving this.

I would be interested in contributing this based upon required effort and my own priorities.

Does the tauri team have any thoughts on how this could be implemented?

I did some research and while there is a safaridriver I don't see a WKWebview driver for selenium

**Comment by FabianLars** [comment]
For visibility, cross-linking the comment about why there is no macos support yet (it's what you said in your last sentence): https://github.com/tauri-apps/tauri/issues/5551#issuecomment-1304348684

> Does the tauri team have any thoughts on how this could be implemented?

Afaik only what's in the [readme](https://github.com/tauri-apps/tauri/tree/dev/tooling/webdriver):
> note: the (probably) items haven't been proof-of-concept'd yet, and if it is not possible to use the listed native webdriver, then a custom implementation will be used that wraps around [wry](https://github.com/tauri-apps/wry).

Though tbh that doesn't really sound like a viable option to me, but i'm also completely out of the loop on that topic so idk...

**Comment by PJUllrich** [comment]
Hey folks, just wanted to add that if you try to run `cargo run tauri-driver` inside your `src-tauri` on macOS today, this is the error you'll see.

In the shell, you'll see a warning:

```
warning: my-app v0.1.0 (/Users/my-user/my-app/src-tauri) ignoring invalid dependency `tauri-driver` which is missing a lib target
```

and in the browser console, you'll see:

```
Failed to load resource: Could not connect to the server.
```

I add this only for folks to find it on Google in case they wonder why the app "seems" to start on macOS, but then doesn't work properly :)

**Comment by vdavid** [comment]
https://v2.tauri.app/develop/tests/webdriver/ says "On desktop, only Windows and Linux are supported due to macOS not having a WKWebView driver tool available.", sounds like a major blocker.

**Comment by FabianLars** [comment]
for what its worth, there's a paid offering by CrabNebula https://docs.crabnebula.dev/plugins/tauri-e2e-tests/

i know how much money and time was spent on that so i wouldn't expect something (F)OSS without Apple (or aother entity) creates something like a WKWebDriver.

**Comment by damageboy** [comment]
I'm obviously out of my depth here, but it seems like safari does support this?
https://developer.apple.com/documentation/safari-developer-tools/macos-enabling-webdriver

**Comment by FabianLars** [comment]
Correct but that really only supports Safari and doesn't help us with WKWebView even though both have the same webkit base.

**Comment by damageboy** [comment]
> Correct but that really only supports Safari and doesn't help us with WKWebView even though both have the same webkit base.

Thanks, I was sure I didn't discover anything here, but wanted to understand the real limitation...

**Comment by danielraffel** [comment]
Building a Tauri app on macOS? I built an open-source WebDriver + MCP integration for real E2E tests so tools (and agents) can click/type/screenshot like they would in a browser. I'm using it while I build my first Tauri app. Worth mentioning this is fresh out of the oven and written with Claude. I wrote about it [here](https://danielraffel.me/2026/02/14/i-built-a-webdriver-for-wkwebview-tauri-apps-on-macos/) and you can find the repo [here](https://github.com/danielraffel/tauri-webdriver) which links to the two crates.

**Comment by srsholmes** [comment]
Hey folks — I ran into the same macOS limitation with tauri-driver and ended up building a different approach: a Tauri plugin that embeds a socket server in the app, letting Playwright communicate with the real webview directly (no WebDriver/CDP needed).

It supports macOS, Windows, and Linux out of the box. There's also a headless browser mode with mocked IPC for fast CI runs.

Repo: https://github.com/srsholmes/tauri-playwright

Just published to npm (`@srsholmes/tauri-playwright`) and crates.io (`tauri-plugin-playwright`). Hope it helps anyone stuck on this!

**Comment by vdavid** [comment]
That's very nice, @srsholmes! I've added it to my app ([Cmdr](https://github.com/vdavid/cmdr)), it works, as in, the tests are passing, and the benchmarks are pretty convincing:
**Before:** I had an awkward Docker+Linux E2E setup that took **10 min** to run, and a CrabNebula beta API key thing which I couldn't even benchmark because that suite apparently broke a week ago and I didn't notice.
**After:** with your lib, the same test suite runs in **45 sec**, a fancy 92.5% speedup if I'm calculating right.

I also need a lot fewer workarounds /hacks than what I needed with the old solutions.

I've sent a [PR](https://github.com/srsholmes/tauri-playwright/pull/1) to add the stuff I was missing.

**Comment by srsholmes** [comment]
Amazing @vdavid, I'm glad you tried it out and got it working. Thanks for trying it out and letting us know!
